DungeonMaster/Assets/Scripts/DeveloperConsole.cs

```diff
@@ -0,0 +1,179 @@
+        } else {
+            GlobalDebug.LogMessage("request succeeded, url: " + www.url + " responseCode: " + www.responseCode + " body: " + www.downloadHandler.text);
+            string json = www.downloadHandler.text;
+            json = "{\"" + currentDataType.ToString() + "\":" + json + "}";
+            SetList(json);
+        }
+    }
+
+    void SetList(string json) {
+        switch (currentDataType) {
+            case DataType.items:
+                itemList = JsonUtility.FromJson<ItemList>(json);
+                ShowItems(typeof(Item), itemList.items);
+                break;
+            case DataType.characters:
+                characterList = JsonUtility.FromJson<CharacterList>(json);
+                ShowItems(typeof(Character), characterList.characters);
+                break;
+        }
+    }
+
+    void ShowItems(Type type, object[] objs) {
+        CreateHeader(type);
+        foreach (var obj in objs) {
+            GameObject horizontalGroup = Instantiate(horizontalLayoutGroup, scrollViewContent.transform);
+            FieldInfo[] fields = GetFields(obj);
+            foreach (FieldInfo field in fields) {
+                GameObject input = Instantiate(inputField, horizontalGroup.transform);
+                input.GetComponent<InputField>().text = field.GetValue(obj).ToString();
+            }
+        }
+    }
+
+    void CreateHeader(Type type) {
+        GameObject horizontalGroup = Instantiate(horizontalLayoutGroup, scrollViewContent.transform);
+
+        FieldInfo[] fields = type.GetFields(BindingFlags.NonPublic | BindingFlags.Instance);
+        foreach (FieldInfo field in fields) {
+            GameObject header = Instantiate(headerText, horizontalGroup.transform);
+            header.GetComponent<Text>().text = field.Name;
+        }
+    }
+
+    FieldInfo[] GetFields(object obj) {
+        Type type = obj.GetType();
+        return type.GetFields(BindingFlags.NonPublic | BindingFlags.Instance);
+    }
+
+    public void AddSelectedType() {
+        CleanScrollView();
+        StartCoroutine(ADDNEW());
+        DropdownValueChanged(databaseSelector);
+    }
+
+    IEnumerator ADDNEW() {
+        string json = GetJSONForType();
+        Debug.Log("JSON " + json);
+
+        byte[] formData = System.Text.Encoding.UTF8.GetBytes(json);
+        UnityWebRequest www = CreatePostRequest(baseURL + currentDataType, formData);
+
+        yield return StartCoroutine(WaitForRequest(www, currentDataType.ToString()));
+    }
+
+    string GetJSONForType() {
+        string json = "";
+
+        switch (currentDataType) {
+            case DataType.items:
+                Item item = new Item(null, "s", "s", "s", "s", 1, new Currency(0, 0, 0, 0, 0), 0);
+                json = JsonUtility.ToJson(item);
+                break;
+            case DataType.characters:
+                Stats stats = new Stats(0, 0, 0, 0, 0, 0);
+                Skills skills = new Skills(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
```

Comment:
So for these that have default values make a static var for the class like Skills.default

Reply:
oooh i was wondering how to do this cleanly